src/client.rs

```diff
@@ -1307,6 +1308,18 @@ impl<T: Read + Write + Unpin + fmt::Debug + Send> Session<T> {
         Ok(c)
     }
 
+    /// The [`ID` command](https://datatracker.ietf.org/doc/html/rfc2971)
+    pub async fn id(&mut self, identification: &str) -> Result<Option<HashMap<String, String>>> {
```

Comment:
this should be `HashMap<String, Option<String>>`, according to the spec the value can be `NIL`

Reply:
what is the format of `identification`, isn't that a list of key values as well?

Reply:
well it would be nice to accept `impl Iter<Item = (&str, Option<&str)>`, that way there is less chance to screw up the formatting